### PR TITLE
[Feature] AddWeighted

### LINKED ISSUE #233: [Feature Request] Implement addWeighted
we might want to implement `addWeighted` from OpenCV.

https://docs.opencv.org/3.4/d2/de8/group__core__array.html#gafafb2513349db3bcff51f54ee5592a19

the motivation is that all can be used as base to implement all the adjust functions

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference and Closes #233

### CODE REVIEW

**Comment by kshitij12345** [comment]
@edgarriba I was wondering that if `alpha,beta,gamma` should be `float` or should they be `Tensor` allowing to have different weight for multiple image pairs in the same batch.

File: kornia/color/core.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    See :class:`~kornia.color.AddWeighted` for details.
+    """
+    if not isinstance(src1, torch.Tensor):
+        raise TypeError("src1 should be a tensor. Got {}".format(type(src1)))
+
+    if not isinstance(src2, torch.Tensor):
+        raise TypeError("src2 should be a tensor. Got {}".format(type(src2)))
+
+    if not isinstance(alpha, float):
+        raise TypeError("alpha should be a tensor or float. Got {}".format(type(alpha)))
```
Comment (ananyamukh6) [inline kornia/color/core.py]: The type hint in line 9 says `alpha`, `beta`, `gamma` are `float`s, and we are checking that it is indeed `float`, but the error message says `tensor` or `float`
Maybe you want to use `typing.Union` in the type hint if you want to to accept `tensor` and `float` and also extend the type check in line 20 to accept `tensor`
or
maybe you need to correct the error message to remove `tensor` from the error string

Comment (kshitij12345) [reply kornia/color/core.py]: Ah. Thanks. I previously intended it to be `Tensor` or `float` but then planned against it.

**Comment by edgarriba** [comment]
@kshitij12345 if you go for accepting raw tensors as input just make sure you test everything

File: kornia/color/core.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        raise TypeError("gamma should be a float. Got {}".format(type(gamma)))
+
+    return src1 * alpha + src2 * beta + gamma
+
+
+class AddWeighted(nn.Module):
+    r"""Calculates the weighted sum of two Tensors.
+
+    The function calculates the weighted sum of two Tensors as follows:
+        out = src1 * alpha + src2 * beta + gamma
```
Comment (edgarriba) [inline kornia/color/core.py]: write this as math

**Comment by kshitij12345** [comment]
> LGTM. Don't forget to add to docs. You can open a new PR if you decide to go for tensor arguments

Sure. Forgot that one. Thanks. 
Will support tensor arguments in future PR.

File: kornia/color/core.py (hunk 2 above)
Comment (kshitij12345) [reply kornia/color/core.py]: Done.

File: kornia/color/core.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+from typing import Union, Tuple
```
Comment (dvd42) [inline kornia/color/core.py]: Seems like its not used.

Comment (kshitij12345) [reply kornia/color/core.py]: Done.

**Comment by edgarriba** [comment]
@dvd42 that's a good point :D 
@kshitij12345 what other contributions do you have in mind ?

**Comment by kshitij12345** [comment]
@edgarriba @dvd42 Sorry for the delayed response. I know the `core` in `color` is deceiving place for this to be. But didn't feel that any other file in `color` was a suitable place for the code either. But as of now we decided to keep it in `color`. We can shift it later as and when needed.

**Pull request merged**